All_home_works/hw8/task2.py

```diff
@@ -0,0 +1,46 @@
+"""
+Description in task2.rst
+"""
+import sqlite3
+
+
+class TableData:
+    def __init__(self, db_name, db_table):
+        self._conn = sqlite3.connect(db_name)
+        self._conn.row_factory = sqlite3.Row
+        self._db_table = db_table
+
+    def check_if_table_in_db(self):
+        self.cursor.execute(
+            "SELECT name FROM sqlite_master WHERE type='table' AND name=?;",
+            (self._db_table,),
+        )
+        if self.cursor.fetchone() is None:
+            raise sqlite3.OperationalError(f"no such table: {self._db_table}")
+
+    def __enter__(self):
+        self.cursor = self._conn.cursor()
+        self.check_if_table_in_db()
+        return self
+
+    def __iter__(self):
+        self.cursor.execute(f"SELECT * FROM '{self._db_table}';")  # noqa: S608
+        return (dict(zip(data.keys(), data)) for data in self.cursor.fetchall())
```

Comment:
What problem could we face here?

Reply:
- A lot of data.
- Something has changed in the data.

Reply:
> A lot of data.

How we can handle it?

Reply:
We can each iteration take a new line from DB. But it's also not so fast as to take whole data. And we will have many requests to DB.

Reply:
Let's do something in the middle.  Make a single request and read a row from DB on each iteration.